### ISSUE TITLE
Overflowing "complex" arithmetic operations of known local integers aren't reported

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust#101863]
```rust
pub fn foo() -> u8 {
    let a: u8 = 253;
    let b: u8 = 1;
    let c: u8 = 2;
    let d: u8 = a + b;
    let e: u8 = d + c;
    e
}

fn main() {
    dbg!(foo());
}
```

Intuitive stuff like `let _ = i32::MAX + 1` is already reported but the above snippet is a trivial example of known local integers that overflow and should be reported as such.

**Comment by c410-f3r** [comment]
Closing due to the lack of activity.
